### ISSUE TITLE
failed to create LLB definition: dockerfile line greater than max allowed size of 65535

### ISSUE THREAD

**Issue by tarun-discovery** [pulumi/pulumi-docker#551]
### What happened?

I am trying to build and push a docker image to ECR with custom tag.
Previously I was using awsx package but it doesnt supports pushing custom tag.

So i am using docker package now but getting errors during pulumi up step

### Expected Behavior

Docker file should have been built and pushed to ECR with custom tag

### Steps to reproduce

Trying to build and push docker image to ECR Repo
Code:
   ```
 const appImage = new docker.Image(namespace, {
            build: {
                context: '../../services/-foldername-redacted-',
                dockerfile: '../../services/-foldername-redacted-/Dockerfile',
                args: {
                    'SERVICE_PORT': containerPort.toString(),
                    'GRPC_PORT': grpcPort.toString()
                }
            },
            imageName: pulumi.interpolate`${repository.repositoryUrl}:test1`,
            registry: authToken,
        });
```

I am getting below error during image building via Pulumi up
` failed to solve with frontend dockerfile.v0: failed to create LLB definition: dockerfile line greater than max allowed size of 65535`

```
Following is the code to get authToken
 const authToken = repository.registryId.apply(async id => {
            const credentials = await aws.ecr.getCredentials({ registryId: id });
            const decodedCredentials = Buffer.from(credentials.authorizationToken, 'base64').toString();
            const [username, password] = decodedCredentials.split(':');
            if (!password || !username) {
                throw new Error('Invalid credentials');
            }
            return {
                server: credentials.proxyEndpoint,
                username: username,
                password: password,
            };
        });
```

### Output of `pulumi about`

CLI          
Version      3.15.0
Go Version   go1.17.2
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       darwin
Version  12.6.3
Arch     x86_64

This project is written in nodejs (/Users/tsinghal/.nvm/versions/node/v14.18.1/bin/node v14.18.1)

Backend  
Name     muml289240m
URL      s3://--redacted--
User     tsinghal

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AaronFriel** [comment]
Hey @tarun-discovery, thanks for flagging this. 

This could be due to the provider parsing an incorrect Dockerfile, which we're tracking in #529 and we expect to fix in the next release.

The error message for a line longer than 65535 characters is surprising though, do you have any very large commands injected into the Dockerfile, or is it possible for you to share a minimal GitHub repository or Gist that we can use to reproduce this?

**Comment by tarun-discovery** [comment]
@AaronFriel thanks for responding, our repo is private, however I am using this docker file
FROM alpine

ARG SERVICE_PORT

EXPOSE $SERVICE_PORT

RUN apk update && apk add ca-certificates && rm -rf /var/cache/apk/*

WORKDIR /usr/service
ADD app ./
RUN addgroup -g 10000 -S app && \
    adduser -u 10000 -S app -G app && \
    chmod +x app

ENTRYPOINT [ "./app" ]

**Comment by AaronFriel** [comment]
Try removing the `*` in `rm rf ...`? It's possible that the glob expansion exceeded shell limits.

**Comment by tarun-discovery** [comment]
ok checking, also as per your 1st comment are you saying that I cant have context in some other directory
```
context: '../../services/gms_kafka_replicator',
                dockerfile: '../../services/gms_kafka_replicator/Dockerfile',
If you noticed Dockerfile is present relative to context directory, its not in some other directory.

Is it that I cant have context in some other directory then the file from where I am executing docker image function?
```

**Comment by tarun-discovery** [comment]
also removing * doesnt help it still throws "dockerfile line greater than max allowed size of 65535"

**Comment by tarun-discovery** [comment]
I even moved dockerfile in same directory and update code like below

```
   const appImage = new docker.Image(namespace, {
            build: {
                context: '.',
                dockerfile: 'Dockerfile',
                args: {
                    'SERVICE_PORT': containerPort.toString(),
                    'GRPC_PORT': grpcPort.toString(),
                },
            },
            imageName: pulumi.interpolate`${repository.repositoryUrl}:test1`,
            // registry: authToken,
        });

```
still same issue

**Comment by tarun-discovery** [comment]
@AaronFriel any news?

**Comment by AaronFriel** [comment]
@tarun-discovery we released 4.0.1 which included the changes in #542, among many other improvements. You should be able to specify a Dockerfile in any location now, including inside the context dir. We also fixed an unintentional break in the `dockerfile` argument in v4, which made it relative to the context dir. 

Please give that a go, and if that doesn't work we may need to talk through Slack to discuss the Dockerfile that's giving you this error, we haven't been able to reproduce it.
